### ISSUE TITLE
error setting up Stages reconciler: index PromotionPolicies by Stage: no matches for kind \"PromotionPolicy\" in version \"kargo.akuity.io/v1alpha1\"

### ISSUE THREAD

**Issue by patrickblackjr** [akuity/kargo#2011]
# Checklist

* [x] I've searched the issue queue to verify this is not a duplicate bug report.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `kargo version`.
* [x] I've pasted logs, if applicable.

# Description

Unable to start using v0.6.0 of the Helm chart via ArgoCD. The API starts successfully, but all other deployments/pods fail for various reasons.

# Screenshots

N/A

# Steps to Reproduce

1. Install using the following ArgoCD Application manifest:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: kargo
  namespace: argocd
spec:
  project: default
  source:
    repoURL: https://github.com/akuity/kargo.git
    targetRevision: v0.6.0
    path: charts/kargo
    helm:
      valuesObject:
        crds:
          keep: false
        managementController:
          logLevel: DEBUG
        controller:
          logLevel: DEBUG
        api:
          logLevel: DEBUG
          adminAccount:
            passwordHash: $2a$10$Zrhhie4vLz5ygtVSaif6o.qN36jgs6vjtMBdM6yrU1FOeiAAMMxOm
            tokenSigningKey: iwishtowashmyirishwristwatch

  destination:
    namespace: kargo
    name: in-cluster
  syncPolicy:
    automated:
      prune: true
      selfHeal: true
      allowEmpty: true
    syncOptions:
      # - ServerSideApply=true
      - CreateNamespace=true
```

2. Wait for ArgoCD to deploy the application.
3. Review the logs.

# Version

```json
{
  "Version": "v0.1.0", (unsure why this is showing when using the above Helm chart version, this may be the problem, but I'm unsure why it's happening)
  "BuildDate": "2023-09-19T19:32:58Z",
  "GitCommit": "9203c78b0b9fc37023a5bb5952a1a0edac2f9bfd",
  "GitTreeDirty": false,
  "GoVersion": "go1.21.0",
  "Compiler": "gc",
  "Platform": "linux/amd64"
}
```

# Logs

```
kargo-controller
time="2024-05-15T03:53:35Z" level=info msg="Starting Kargo Controller" commit=9203c78b0b9fc37023a5bb5952a1a0edac2f9bfd version=v0.1.0
                                 time="2024-05-15T03:53:35Z" level=debug msg="loading in-cluster REST config"
                                 time="2024-05-15T03:53:35Z" level=debug msg="loading in-cluster REST config"
                                 I0515 03:53:36.882304       1 request.go:601] Waited for 1.144583489s due to client-side throttling, not priority and fairness, request: GET:https://10.4.0.1:443/apis/apiextensions.crossplane.io/v1?timeout=32s
                                 time="2024-05-15T03:53:43Z" level=error msg="error setting up Stages reconciler: index PromotionPolicies by Stage: no matches for kind \"PromotionPolicy\" in version \"kargo.akuity.io/v1alpha1\""

kargo-management-controller
time="2024-05-15T03:52:45Z" level=error msg="unknown command \"management-controller\" for \"kargo\""

kargo-webhooks-server
time="2024-05-15T03:53:58Z" level=info msg="Starting Kargo Webhooks Server" commit=9203c78b0b9fc37023a5bb5952a1a0edac2f9bfd version=v0.1.0
I0515 03:53:59.229757       1 request.go:601] Waited for 1.072870014s due to client-side throttling, not priority and fairness, request: GET:https://10.4.0.1:443/apis/apiextensions.crossplane.io/v1alpha1?timeout=32s
time="2024-05-15T03:54:06Z" level=error msg="index PromotionPolicies by Stage: no matches for kind \"PromotionPolicy\" in version \"kargo.akuity.io/v1alpha1\""

===kargo-api=== [THIS POD IS STARTING SUCCESSFULLY] 
time="2024-05-15T03:36:30Z" level=info msg="Starting Kargo API Server" commit=9203c78b0b9fc37023a5bb5952a1a0edac2f9bfd version=v0.1.0
time="2024-05-15T03:36:30Z" level=debug msg="loading in-cluster REST config"
time="2024-05-15T03:36:30Z" level=info msg="admin account is enabled"
time="2024-05-15T03:36:34Z" level=info msg="Server is listening on \"[::]:8080\"" tls=true
2024/05/15 03:37:49 http: TLS handshake error from 127.0.0.1:46886: EOF
2024/05/15 03:37:49 http: TLS handshake error from 127.0.0.1:46896: EOF
```

**Comment by krancour** [comment]
This is a side effect of having installing from chart _source_ instead of installing from the chart we officially publish in our charts repo [here](https://github.com/akuity/kargo/pkgs/container/kargo-charts%2Fkargo).

The `Chart.yaml` file uses `v0.1.0` as a placeholder for `appVersion` (because v0.0.0 and non-semver values are not permitted). This value is overridden automatically when we build and publish the chart at release time.

The version of the Kargo image that is used is, by default, derived from the `appVersion` -- so you're ending up with manifests (including CRDs) from `v0.6.0`, but its being paired with an _ancient_ `v0.1.0` image. That image is expecting the existence of a `PromotionPolicy` CRD in the cluster. The CRD has long since been removed from the project, with its responsibilities roled into the `Project` CRD.

Bottom line is, please don't install from here:

```
source:
    repoURL: https://github.com/akuity/kargo.git
    targetRevision: v0.6.0
    path: charts/kargo
```

Please do this instead:

```
source:
    repoURL: ghcr.io/akuity/kargo-charts
    chart: kargo
    targetRevision: 0.6.0
```
